api/http/api_inventory.go

```diff
@@ -588,6 +598,69 @@ func (i *inventoryHandlers) CreateTenantHandler(w rest.ResponseWriter, r *rest.R
 	w.WriteHeader(http.StatusCreated)
 }
 
+func (i *inventoryHandlers) FiltersSearchHandler(w rest.ResponseWriter, r *rest.Request) {
+	ctx := r.Context()
+
+	l := log.FromContext(ctx)
+
+	//extract attributes from body
+	searchParams, err := parseSearchParams(r)
+	if err != nil {
+		u.RestErrWithLog(w, r, l, err, http.StatusBadRequest)
+		return
+	}
+
+	// query database
+	devs, totalCount, err := i.inventory.SearchDevices(ctx, *searchParams)
+	if err != nil {
+		if strings.Contains(err.Error(), "BadValue") {
+			u.RestErrWithLog(w, r, l, err, http.StatusBadRequest)
+		} else {
+			u.RestErrWithLogInternal(w, r, l, err)
+		}
+		return
+	}
+
+	// the response writer will ensure the header name is in Kebab-Pascal-Case
+	w.Header().Add(hdrTotalCount, strconv.Itoa(totalCount))
+	w.WriteJson(devs)
+}
+
+func (i *inventoryHandlers) InternalFiltersSearchHandler(w rest.ResponseWriter, r *rest.Request) {
+	ctx := r.Context()
+
+	l := log.FromContext(ctx)
+
+	tenantId := r.PathParam("tenant_id")
+	if tenantId == "" {
+		u.RestErrWithLog(w, r, l, errors.New("tenant_id not provided"), http.StatusBadRequest)
+		return
+	}
+	ctx = getTenantContext(ctx, tenantId)
+
+	//extract attributes from body
+	searchParams, err := parseSearchParams(r)
+	if err != nil {
+		u.RestErrWithLog(w, r, l, err, http.StatusBadRequest)
+		return
+	}
+
+	// query database
+	devs, totalCount, err := i.inventory.SearchDevices(ctx, *searchParams)
+	if err != nil {
+		if strings.Contains(err.Error(), "BadValue") {
+			u.RestErrWithLog(w, r, l, err, http.StatusBadRequest)
```

Comment:
adding this unit test scenario will make codecov happy.

Reply:
done